app/src/main/java/com/vietdung/mvparchitect/MainActivity.java

```diff
@@ -1,13 +1,79 @@
 package com.vietdung.mvparchitect;
 
-import android.support.v7.app.AppCompatActivity;
+import android.app.SearchManager;
+import android.content.Context;
+import android.content.Intent;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import java.util.ArrayList;
+import java.util.List;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends AppCompatActivity implements onTaskListerner, SearchUserContract.View {
+
+    private RecyclerView mRecyclerView;
+    private List<User> mUsers;
+    private UserAdapter mUserAdapter;
+    private SearchView mSearchView = null;
+    private SearchUserPresenter mSearchUserPresenter;
```

Comment:
Không khai báo trực tiếp class implement mà nên khai báo interface
```suggestion
    private SearchUserContract.Presenter mPresenter;
```